### DISCUSSION TITLE
[Frequently Asked] Exported PPTX cannot re-edit

### DISCUSSION THREAD

**Discussion by yhatt** [marp-team/marp#82] [Frequently Asked]
PPTX converted in Marp (CLI / VS Code) has consisted by "pre-rendered slide images".

This output is sufficient to play your slide / add annotations on a familiar tool, but not suitable to handle text content / re-edit the content.

## Why

Marp team's requirement to the converted output is almost 100% reproducibility to the slide design, and pre-rendered image is nearly perfect in regards to this point. We think it's too difficult to make an editable PPTX output with keeping the most part of complex HTML/CSS based layout.

## Solutions

If you don't care about slightly less design reproducibility of the slide, you can convert editable PPTX.

### Solution 1: Enable experimental option ([Marp CLI >= v4.1.0](https://github.com/marp-team/marp-cli/releases/tag/v4.1.0) / [Marp for VS Code >= v3.1.0](https://github.com/marp-team/marp-vscode/releases/tag/v3.1.0))

You have to install both of browser and [LibreOffice Impress](https://www.libreoffice.org/).

### Marp CLI

```bash
marp --pptx --pptx-editable your-slide.md
```

### Marp for VS Code

Set `markdown.marp.pptx.editable` preference as `true`.

> [!WARNING]
>
> This is an experimental option and still incomplete. Some slides may throw an error or output the incomplete result. (e.g. Using `gaia` theme)

### Solution 2: PDF to PPTX

You can get editable PPTX by converting exported PDF to PPTX too.

- **[Adobe Acrobat: PDF to PPTX](https://www.adobe.com/acrobat/online/pdf-to-ppt.html)**

#### Related issues 

- marp-team/marp-cli#166
- marp-team/marp-cli#298

**Comment by Beurt** [comment]
Hi,

Thank you very much for this excellent feature!

However, I tried with marp-cli 4.1.1 (freshly downloaded) and cannot manage to make it work. It makes a PPTX, but as usual with a background image rendered form the HTML slide. Hence, not editable. I am using Linux and LO Impress is installed of course.

My command line was like:

```bash
marp --pptx --pptx-editable --theme Mytheme.css myfile.md -o test.pptx
```

**Reply by yhatt** [reply]
You have set `--pptx-editable` option so Marp CLI should try to convert into editable PPTX. I have tried reproduce with a described command, but Marp CLI on my Ubuntu has tried to convert editable PPTX.

Set additional `--debug=true` option to peek whether actually Marp CLI is trying to convert into editable PPTX.

**Reply by Beurt** [reply]
Thanks for your tests and your reply.

I'm using an Ubuntu too (24.10) with the downloaded binary of marp-cli. The debug output is:

```bash
~/marp --pptx --pptx-editable --allow-local-files test.md -o test.pptx --debug=true
  marp-cli Using a polyfilled implementation for Intl.Segmenter. +0ms
[  INFO ] Debug logging is enabled. (Filter pattern: marp-cli*)
  marp-cli:engine Failed to require @marp-team/marp-core. +0ms
  marp-cli:engine Error: Cannot find module '@marp-team/marp-core'
  marp-cli:engine Require stack:
  marp-cli:engine - /mypath/R2.03 Ergonomie et Accessibilite/noop.js
  marp-cli:engine 1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
  marp-cli:engine     at Function.<anonymous> (node:internal/modules/cjs/loader:1253:15)
  marp-cli:engine     at Function._resolveFilename (pkg/prelude/bootstrap.js:1959:46)
  marp-cli:engine     at Function._load (node:internal/modules/cjs/loader:1079:27)
  marp-cli:engine     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
  marp-cli:engine     at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
  marp-cli:engine     at Module.<anonymous> (node:internal/modules/cjs/loader:1344:12)
  marp-cli:engine     at Module.require (pkg/prelude/bootstrap.js:1855:31)
  marp-cli:engine     at require (node:internal/modules/helpers:141:16)
  marp-cli:engine     at /snapshot/marp-cli/lib/marp-cli-Ccx5FcVa.js:18:89851
  marp-cli:engine     at _o._silentRequire (/snapshot/marp-cli/lib/marp-cli-Ccx5FcVa.js:18:110529) {
  marp-cli:engine   code: 'MODULE_NOT_FOUND',
  marp-cli:engine   requireStack: [
  marp-cli:engine     '/myPath/R2.03 Ergonomie et Accessibilite/noop.js'
  marp-cli:engine   ],
  marp-cli:engine   pkg: true
  marp-cli:engine } +0ms
  marp-cli:browser Browser manager configured: { finders: undefined, path: undefined, protocol: 'cdp', timeout: undefined } +0ms
[  INFO ] Converting 1 markdown...
  marp-cli:browser:finder Start finding browser from chrome, edge, firefox ({ preferredPath: undefined }) +0ms
  marp-cli:browser:finder Found chrome: { path: '/usr/bin/chromium', acceptedBrowsers: [ [class nn extends Pt] { kind: 'chrome', protocol: 'webDriverBiDi' }, [class rn extends nn] { protocol: 'cdp' } ] } +26ms
  marp-cli:browser:finder Use browser: { path: '/usr/bin/chromium', acceptedBrowsers: [ [class nn extends Pt] { kind: 'chrome', protocol: 'webDriverBiDi' }, [class rn extends nn] { protocol: 'cdp' } ] } +0ms
  marp-cli:browser Use browser class for conversion: [class rn extends nn] { protocol: 'cdp' } +30ms
[  WARN ] Insecure local file accessing is enabled for conversion from test.md.
  marp-cli CLIError: Edge browser could not be found.
  marp-cli     at fe (/snapshot/marp-cli/lib/marp-cli-Ccx5FcVa.js:1:8198)
  marp-cli     at edge (/snapshot/marp-cli/lib/marp-cli-Ccx5FcVa.js:18:57201) {
  marp-cli   errorCode: 2
  marp-cli } +459ms
  marp-cli Saving temporary file: /pathtotmp/tmp-123206-bse6HCPiInJ6-.html +0ms
  marp-cli Saving file: /pathtotmp/tmp-123206-bse6HCPiInJ6-.html +0ms
  marp-cli Saved: /pathtotmp/tmp-123206-bse6HCPiInJ6-.html +2ms
  marp-cli:browser Chrome data directory: /tmp/marp-cli-KeI-hmFrwS +83ms
  marp-cli:browser Created data directory: /tmp/marp-cli-KeI-hmFrwS +0ms
  marp-cli Cleaned up temporary file: /pathtotmp/tmp-123206-bse6HCPiInJ6-.html +1s
  marp-cli Saving file: test.pptx +9ms
  marp-cli Saved: test.pptx +2ms
[  INFO ] test.md => test.pptx
  marp-cli:browser Browser disconnected (Cleaned up puppeteer instance) +1s
```

And the output pptx file is, unfortunately, not editable.

**Reply by yhatt** [reply]
Let me confirm, just to be sure: What is the output of `marp --version` ?

This output indicates that `--pptx-editable` option has not been recognized. If recognized, Marp CLI will output a notice about using experimental option.

**Reply by Beurt** [reply]
You are absolutely right, the error is on my part!

I thought I was using the latest version, but, because of a wrong copy/paste of the binary, I was using an older version.

With v4.1.1, it works perfectly. Thank you a lot!
